### PR TITLE
feat(APIv2): RHINENG-1986 branch out historical test results

### PR DESCRIPTION
**Pull request by skateman**
The `V2::TestResult` model now returns only what `::TestResult.latest` is returning and the rest of test results are available under `V2::HistoricalTestResults`. For now we don't have multiple test results for a single tailoring/system combo, but the codebase always allowed it. 

This should allow us in the future to compare historical test results in a similar fashion as Drift does with system profiles.

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -408,6 +408,26 @@
...
+      test_results.supported,
+      test_results.failed_rule_count,
+      test_results.created_at,
+      test_results.updated_at
+     FROM ((test_results
+       JOIN profiles ON ((profiles.id = test_results.profile_id)))
+       JOIN ( SELECT test_results_1.profile_id,
+              test_results_1.host_id,
+              max(test_results_1.end_time) AS end_time
+             FROM test_results test_results_1
```
Comment (vkrizan) [inline db/schema.rb:428]: add `AS` in-between to make the syntax consistent.

File: db/schema.rb [hunk 2]
```diff
@@ -408,6 +408,26 @@
...
+            GROUP BY test_results_1.profile_id, test_results_1.host_id) tr ON (((test_results.profile_id = tr.profile_id) AND (test_results.host_id = tr.host_id) AND (test_results.end_time = tr.end_time))));
```
Comment (vkrizan) [inline db/schema.rb:429]: The same here for the `tr`.

Why not join on primary key here?

Comment (skateman) [reply db/schema.rb:429]: This is generated by `pg_dump`, look at the SQL files lower.

**Pull request merged**